pymc/distributions/distribution.py

```diff
@@ -178,11 +185,55 @@ def _make_rv_and_resize_shape_from_dims(
     return rv_out, resize_shape_from_dims
 
 
+class SymbolicRandomVariable(OpFromGraph):
+    """Symbolic Random Variable
+
+    This is a subclasse of `OpFromGraph` which is used to encapsulate the symbolic
+    random graph of complex distributions which are built on top of pure
+    `RandomVariable`s.
+
+    These graphs may vary structurally based on the inputs (e.g., their dimensionality),
+    and usually require that random inputs have specific shapes for correct outputs
+    (e.g., avoiding broadcasting of random inputs). Due to this, most distributions that
+    return SymbolicRandomVariable create their these graphs at runtime via the
+    classmethod `cls.rv_op`, taking care to clone and resize random inputs, if needed.
+    """
+
+    ndim_supp: int = None
+    """Number of support dimensions as in RandomVariables
+    (0 for scalar, 1 for vector, ...)
+     """
+
+    inline_aeppl: bool = False
+    """Specifies whether the logprob function is derived automatically by introspection
+    of the inner graph.
+
+    If `False`, a logprob function must be dispatched directly to the subclass type.
+    """
+
+    _print_name: Tuple[str, str] = ("Unknown", "\\operatorname{Unknown}")
+    """Tuple of (name, latex name) used for for pretty-printing variables of this type"""
+
+    def __init__(self, *args, ndim_supp, **kwargs):
+        self.ndim_supp = ndim_supp
+        kwargs.setdefault("inline", True)
+        super().__init__(*args, **kwargs)
+
+    def update(self, node: Node):
+        """Symbolic update expression for input random state variables
+
+        Returns a dictionary with the symbolic expressions required for correct updating
+        of random state input variables repeated function evaluations. This is used by
+        `aesaraf.compile_pymc`.
+        """
+        return {}
+
+
 class Distribution(metaclass=DistributionMeta):
     """Statistical distribution"""
 
-    rv_class = None
-    rv_op: RandomVariable = None
+    rv_op: [RandomVariable, SymbolicRandomVariable] = None
+    rv_type: MetaType = None
```

Comment:
What is the need for `rv_type`? I must be missing something...

Here is my understanding: It seems that non-symbolic distributions still get an `rv_op` but are assigned an `rv_type` when calling `Distribution`, but symbolic distributions need a `rv_op` classmethod to be manually created *without* (? not sure here) `rv_type`.

Reply:
The rv_type is not necessary but it was useful for `test_all_distributions_have_moments`. Vanilla distributions `rv_type` is added automatically by the metaclass, it's simply the `type(rv_op)`.

This seems useful to know because it's the type on which all methods are dispatched: logp/ logcdf/ moment/ change_dist_size